datahub-web-react/src/app/search/SearchPageHeader.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { useHistory } from 'react-router';
+import { Typography, Image, Space, AutoComplete, Input, Row } from 'antd';
+import { ManageAccount } from '../shared/ManageAccount';
+import { useGetAuthenticatedUser } from '../useGetAuthenticatedUser';
+import { GlobalCfg, SearchCfg } from '../../conf';
+import { useEntityRegistry } from '../useEntityRegistry';
+import { navigateToSearchUrl } from './utils/navigateToSearchUrl';
+import { useGetAutoCompleteResultsLazyQuery } from '../../graphql/search.generated';
+
+const styles = {
+    background: {
+        width: '100%',
+        backgroundImage: 'linear-gradient(#132935, #FFFFFF)',
+    },
+    navBar: { padding: '24px' },
+    welcomeText: { color: '#FFFFFF', fontSize: 16 },
+    searchContainer: { width: '100%', marginTop: '40px', marginBottom: '160px' },
+    logoImage: { width: 140 },
+    searchBox: { width: 540, margin: '40px 0px' },
+    subHeaderText: { color: '#FFFFFF', fontSize: 20 },
+};
+
+export const SearchPageHeader = () => {
+    const history = useHistory();
+    const entityRegistry = useEntityRegistry();
+    const { data } = useGetAuthenticatedUser();
+    const [getAutoCompleteResults, { data: suggestionsData }] = useGetAutoCompleteResultsLazyQuery();
+
+    const onSearch = (query: string) => {
+        navigateToSearchUrl({
+            query,
+            history,
+            entityRegistry,
+        });
+    };
+
+    const onAutoComplete = (query: string) => {
+        getAutoCompleteResults({
+            variables: {
+                input: {
+                    type: entityRegistry.getDefaultSearchEntityType(),
+                    query,
+                },
+            },
+        });
+    };
+
+    return (
+        <Space direction="vertical" style={styles.background}>
+            <Row justify="space-between" style={styles.navBar}>
+                <Typography.Text style={styles.welcomeText}>
+                    Welcome back, <b>{data?.corpUser?.info?.firstName || data?.corpUser?.username}</b>.
```

Comment:
should we have a double fallback to skip this line if neither is present? or are we pretty sure this will always resolve to something?

Reply:
Theoretically you should have a username if not a name (it's required for the user's primary key)